core/ast/src/scope.rs

```diff
@@ -2,23 +2,48 @@
 //!
 //! Scopes are used to track the bindings of identifiers in the AST.
 
+use bitflags::bitflags;
 use boa_string::JsString;
 use std::{
     cell::{Cell, RefCell},
     fmt::Debug,
     rc::Rc,
 };
 
+bitflags! {
+    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+    struct BindingFlags: u8 {
+        const MUTABLE  = 1 << 0;
+        const LEX      = 1 << 1;
+        const STRICT   = 1 << 2;
+        const ESCAPES  = 1 << 3;
+        const ACCESSED = 1 << 4;
+    }
+}
+
+impl BindingFlags {
+    fn is_mutable(self) -> bool {
+        self.contains(BindingFlags::MUTABLE)
+    }
+    fn is_lex(self) -> bool {
+        self.contains(BindingFlags::LEX)
+    }
+    fn is_strict(self) -> bool {
+        self.contains(BindingFlags::STRICT)
+    }
+    fn is_escapes(self) -> bool {
```

Comment:
```suggestion
    fn escapes(self) -> bool {
```

I think we should rename this to just `escapes`, `is_escapes` sounds odd.